apps/customer-portal/backend-v2/internal/handler/instances.go

```
@@ -47,13 +47,37 @@ type entityInstanceClient interface {
+// production (main.go calls this unconditionally, no kill switch) — see
+// ProjectHandler.SetCallerScope for why this is a setter rather than a
+// constructor parameter.
+func (h *InstanceHandler) SetCallerScope(resolver *CallerScopeResolver) {
+	h.callerScope = resolver
+}
+
+// checkProjectScope requires caller membership only when scope is a
+// project-scoped filter (scope.projectIDs set — the other two ID slices are
+// always nil by construction, see instanceIDFilters); a no-op for the
+// deployment- and deployed-product-scoped variants.
+func (h *InstanceHandler) checkProjectScope(w http.ResponseWriter, r *http.Request, scope instanceIDFilters, userID, email string) bool {
```

golangci-lint (2.12.2)

[error] func (*InstanceHandler).checkProjectScope is unused

(unused)